### PR TITLE
Fix shot count tracking

### PR DESCRIPTION
**Pull request by mes258**
When a player changes their loadout type, the number of shots would also reset to the max value for that type. This change adds an object to track the number of shots left per type. When the player cycles through the different loadouts, the number of shots left does not reset. 

This means each player has a finite number of shots available, so we should probably decide on the mechanic to let players restock soon.

### CODE REVIEW

File: client/player.js [hunk 1]
```diff
@@ -26,10 +27,21 @@ var Player = function (position_, world_) {
     name: "scout",
     reloadTime:100,
     loadStatus:100,
-    magazine:300,
+    magazine:50,
     maxMagazine:300,
   };
 
+  // These are the starting values for each type. Players can gather more shots up to the limit for that type.
```
Comment (vinport16) [inline client/player.js]: client should not be responsible for keeping track of this. the server should manage this

File: client/player.js [hunk 2]
```diff
@@ -76,6 +88,8 @@ var Player = function (position_, world_) {
 
   this.setLoadout = function(l){
     loadout = l;
+    loadout.magazine = shotsLeft[loadout.name]
```
Comment (vinport16) [inline client/player.js]: the number of shots left is already stored in loadout.magazine

**Review by vinport16 — CHANGES_REQUESTED** [review]
The server should be managing how many shots the client has left. client should have that info locally but whenever they change class they should be informed how many shots they have for that loadout

File: client/player.js (hunk 2 above)
Comment (mes258) [reply client/player.js]: Each time the loadout is changed, a new instance of the loadout is created, which re-sets the shots left count. I think we either need to store an instance of each loadout type for each client, or store the shots left per client.

File: client/player.js (hunk 1 above)
Comment (mes258) [reply client/player.js]: Moved to server.

File: client/comms/server.js [hunk 3]
```diff
@@ -18,6 +18,26 @@ var Client = function (id_, conn_) {
   this.isTeleporting = false;
   this.loadoutIDX = 0;
   this.loadout = new Loadout(Loadout.SCOUT);
+  //TODO: Remove the advanced loadouts when we have a way to earn the in-game
+  this.unlockedLoadouts = [
```
Comment (vinport16) [inline client/comms/server.js]: maybe this should be a list of instantiated loadout objects that each store the number of shots left

File: client/index.html [hunk 4]
```diff
@@ -25,7 +25,7 @@
         <tr><td><input id="chat-input" type="text"></input></td></tr>
     </table>
     <div id="reload-bar"><div id="bar-container"><div id="status-bar"></div></div></div>
-    <div id="snowballCount">20</div>
+    <div id="snowballCount">50</div>
```
Comment (vinport16) [inline client/index.html]: why did you change this ?

File: client/player.js [hunk 5]
```diff
@@ -26,7 +26,7 @@ var Player = function (position_, world_) {
     name: "scout",
     reloadTime:100,
     loadStatus:100,
-    magazine:300,
+    magazine:50,
```
Comment (vinport16) [inline client/player.js]: the server immediately sets the player's loadout when they join so these numbers don't matter

File: client/comms/server.js (hunk 3 above)
Comment (mes258) [reply client/comms/server.js]: Updated

File: client/index.html (hunk 4 above)
Comment (mes258) [reply client/index.html]: From a previous change, reverted.

File: client/player.js (hunk 5 above)
Comment (mes258) [reply client/player.js]: From a previous change, reverted.

**Comment by mes258** [comment]
Rebased from main and updated the new client file as needed.

File: client/comms/server/client.js [hunk 6]
```diff
@@ -16,7 +16,18 @@ var Client = function (id_, conn_) {
   this.removed = false;
   this.isTeleporting = false;
   this.loadoutIDX = 0;
-  this.loadout = new Loadout(Loadout.SCOUT);
+  //TODO: Remove the advanced loadouts when we have a way to earn the in-game
+  this.unlockedLoadouts = [
```
Comment (vinport16) [inline client/comms/server/client.js:21]: ok how about this: the inventory of available loadouts is private. client has methods for adding, removing, and maybe resetting (?) loadouts, as well as a method that picks the next loadout (so the server doesn't have to think about the implementation of the loadout list

**Review by vinport16 — APPROVED** [review]

File: client/comms/server/client.js (hunk 6 above)
Comment (mes258) [reply client/comms/server/client.js:21]: Updated to add the loadout methods in the client class. Also, now the shot counts will reset at the start of each round.

**Review by vinport16 — APPROVED** [review]
